CFC_WebApp/api/cfc_webapp.py

```diff
@@ -351,15 +353,24 @@ def after_request():
 # This should only be used by createUserProfile since we may not have a UUID
 # yet. All others should use the UUID.
 def verifyUserToken(token):
-  constructedURL = ("https://www.googleapis.com/oauth2/v1/tokeninfo?id_token=%s"%token)
-  r = requests.get(constructedURL)
-  tokenFields = json.loads(r.content)
-  in_client_key = tokenFields['audience']
-  if (in_client_key != client_key):
-    if (in_client_key != hack_client_key):
-      abort(401, "Invalid client key %s" % in_client_key)
-  logging.debug("Found user email %s" % tokenFields['email'])
-  return tokenFields['email']
+    try:
+        # attempt to validate token on the client-side
+        logging.debug("Using OAuth2Client to verify id token")
+        tokenFields = oauth2client.client.verify_id_token(token,client_key)
+        print tokenFields
```

Comment:
This print is not really necessary and should really be removed because it will clutter up the console. However, this change has already taken so long to get through that I won't insist on it. I'll remove it myself before I deploy the changes on the real server.